lib/flagsmith/engine/core.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'semantic'
+require 'securerandom'
+
+require_relative 'environments/models'
+require_relative 'features/models'
+require_relative 'identities/models'
+require_relative 'organisations/models'
+require_relative 'projects/models'
+require_relative 'segments/evaluator'
+require_relative 'segments/models'
+require_relative 'utils/hash_func'
+
+module Flagsmith
+  module Engine
+    # Flags engine methods
+    module Core
+      include Flagsmith::Engine::Segments::Evaluator
+
+      # rubocop:disable Metrics/CyclomaticComplexity
+      def get_identity_feature_states_dict(environment, identity, override_traits = nil)
```

Comment:
Is there a way to mark this method as private? I don't think it should need to be exposed, right?